libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/camera/CameraViewModel.kt

```diff
@@ -1,33 +1,77 @@
 package com.mapbox.navigation.dropin.component.camera
 
 import android.location.Location
-import androidx.lifecycle.ViewModel
-import androidx.lifecycle.viewModelScope
+import androidx.lifecycle.asFlow
 import com.mapbox.geojson.Point
 import com.mapbox.maps.CameraOptions
-import com.mapbox.maps.EdgeInsets
-import com.mapbox.maps.plugin.animation.MapAnimationOptions
-import kotlinx.coroutines.flow.Flow
-import kotlinx.coroutines.flow.MutableSharedFlow
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.dropin.component.location.LocationBehavior
+import com.mapbox.navigation.dropin.lifecycle.UIViewModel
+import kotlinx.coroutines.flow.collect
 import kotlinx.coroutines.launch
 
-// This might be temporary just to test some camera stuff
-internal class CameraViewModel : ViewModel() {
+sealed class CameraAction {
+    data class UpdateLocation(val location: Location) : CameraAction()
```

Comment:
nit: Seems like this should go into the LocationViewModel 🤔

Reply:
Yeah that's why I think this should be merged first and then you could take a look at it in https://github.com/mapbox/mapbox-navigation-android/pull/5533

Reply:
I know the solution to this. We just need to remove this action and `onAttached` completely from `CameraViewModel`. Make the `CamreraComponent` also dependent on `LocationViewModel`

@kmadsen